### ISSUE TITLE
Autoassign Date

### ISSUE THREAD

**Issue by mirisbowring** [eikek/docspell#1415]
Hi,

i did not find a doc reference for this.
After uploading any document, docspell detects and proposes the issue date of the document (this is working great so far).

Is there an option to auto add this date during consumption (or choose the first, id multiple are detected)? 

After i would only need to check the new documents and correct the false dates instead of touching every file to assign the detected date.

**Comment by eikek** [comment]
It's true that the date handling may be a bit confusing. The due-date will be set automatically, because very early there was the "notify me on due items" functionality. And I also wanted to be notified even for not-confirmed items (because I sometimes fix it days later) - which might be wrong, but better a false alarm once in a while.

The item date, however, is not set automatically, because it will cause the item to be sorted to a different position. This could be even more confusing, I think. Now with the dashboard, it is a bit less of a concern maybe. But since you need to look at each item anyways to confirm, it's just one click to set the correct date usually. This click is only saved, if the first date is the correct one.

But I might not understand the issue good enough? You would need to go through all new documents to correct other data as well, anyways - no?

**Comment by mirisbowring** [comment]
My current workflow (coming from paperless-ng) is a bit different.

Imagin, you are using the system for a year already. Therefore you have configured many regex for Tags, created your folder structure and have saved many people and correspondents.

Now, you have like 20 documents, that need to be "imported". The best case is, that no new correspondent is there.

So you scan them, paperless processes them (autotagging, automoving to correct folder and autoassigning new date).
Later, if the user is home and has some free minutes, he just want to check the dashboard of the newly uploaded documents and check (visually) if everything went as expected - He has to click nothing.

This would take some seconds only and he would not click anything else. Currently, he would need to step trough every document.

This would help going from semi automated processing to automated processing. Paperless e.x. does not propose multiple dates (and change them via one click). It's more like a "friss oder stirb" automation and i would need to select the new date via date picker, but 95% of the time, the first date detected, is correct.

**Comment by eikek** [comment]
I don't think this workflow is supported. Docspell wants you to confirm the metadata, because the guessing might not be correct. Of course, the goal is that you only click "Confirm". You could select all 20 and then click confirm, to make it faster. But sometimes a bit adjustment is necessary and not everything can be extracted automatically yet. So it's not possible to go fully automated out of the box. You could create scripts that auto-approve everything periodically…. Also, just to mention it: it is not possible to define regexes or create a nested folder structure.

**Comment by mirisbowring** [comment]
But Correspondent and concerning are already auto-added (besides the proposal). So why not trying to autoassign the date?

Even if the correctness is about 50% - these are 50% less clicks i have to make and I probably can filter documents by date, even if they have not been "confirmed" yet :)

**Comment by mirisbowring** [comment]
> You could create scripts that auto-approve everything periodically…

To clarify: I like the approach of approving new Documents (better than in paperless). But i would like to speed up / simplify the process by prefilling most metadata as possible. So even a document with auto-added date has to be approved.

**Comment by eikek** [comment]
> But Correspondent and concerning are already auto-added (besides the proposal). So why not trying to autoassign the date?
> ...
> **EDIT:** even the due date is automatically estimated and prefilled. :)

I think I already wrote about the reasons [above](https://github.com/eikek/docspell/issues/1415#issuecomment-1049242068). It could happen that you have new documents but they are "invisible" unless you click "Inbox" in the search menu. I think this can be very confusing. Anyway, that's why it is how it is 🤷🏼 

Now there is a dashboard where you could list these as well, but that was just added in the latest version. I don't mind revisiting this again. Not sure if it saves so much time, though. Could be a config option
